cmd/vic-machine/common/certificate.go

```diff
@@ -0,0 +1,408 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"gopkg.in/urfave/cli.v1"
+
+	"github.com/vmware/vic/pkg/certificate"
+	"github.com/vmware/vic/pkg/errors"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+// CertSeed has all input parameters for vic-machine certificate commands
+type CertSeed struct {
+	CertPath              string
+	DisplayName           string
+	Scert                 string
+	Skey                  string
+	Ccert                 string
+	Ckey                  string
+	Cacert                string
+	Cakey                 string
+	ClientCert            *tls.Certificate
+	ClientCAsArg          cli.StringSlice `arg:"tls-ca"`
+	ClientCAs             []byte
+	EnvFile               string
+	Cname                 string
+	Org                   cli.StringSlice
+	KeySize               int
+	NoTLS                 bool
+	NoTLSverify           bool
+	ClientNetworkName     string
+	ClientNetworkIP       string
+	PublicNetworkName     string
+	PublicNetworkIP       string
+	ManagementNetworkName string
+	ManagementNetworkIP   string
+	KeyPEM                []byte
+	CertPEM               []byte
+	Debug                 int
```

Comment:
That's fine, but this will require quite a bit of a refactoring effort that I wanted to avoid with regard to `processCertificates` which relies on those fields.